src/test/resources/field/registerFieldsObject.json

```diff
@@ -1,12 +1,23 @@
 {
   "indexName": "test_index",
   "field": [
+    {
+      "name": "real_id",
+      "type": "_ID",
+      "search": true,
+      "store": true
+    },
     {
       "name": "doc_id",
       "type": "ATOM",
       "search": true,
       "storeDocValues": true
     },
+    {
+      "name": "_nested_type",
```

Comment:
currently I require the user to add these 2 fields(_nested_type, _nested_path) in the schema explicitly. I can hide this to the user. Wondering which approach is preferred. I feel some query like empty query may end up return child docs. The user can use _nested_type to indicates they only want parent docs

Reply:
I think ES automatically adds a filter to the base query so that only root documents are considered, with child document fields returned in inner hits.

If we decide we want to add the ability to query child fields directly, we would probably want it to be a request level option to specify the nested path to filter by. This would avoid needing to expose these metafields in the index schema.